core/src/main/java/org/infinispan/persistence/spi/NonBlockingStore.java

```diff
@@ -80,114 +146,344 @@
        * SEGMENTABLE will never be allowed to be configured as segmented.
        * <p>
        * Stores that have this characteristic must override the {@link #addSegments(IntSet)} and
-       * {@link #removeSegments(IntSet)} methods.
+       * {@link #removeSegments(IntSet)} methods. If a store is {@link #SHAREABLE} and is configured to be shared
+       * via configuration these methods will not be invoked though.
        */
       SEGMENTABLE,
       /**
-       * If this store supports storing expiration metadata. Certain methods may or may not include expired entries.
+       * If this store supports storing expiration metadata. That is this store should never return an expired entry
+       * via any methods such as {@link #load(int, Object)}, {@link #publishKeys(IntSet, Predicate)} or
+       * {@link #publishEntries(IntSet, Predicate, boolean)}. It is recommended that a store use the provided
+       * {@link org.infinispan.commons.time.TimeService} in the {@code InitializationContext} to determine if an
+       * entry has expired.
+       * <p>
+       * The information about an entry and its expiration is included in the {@link org.infinispan.metadata.Metadata}
+       * which is accessible from the {@link MarshallableEntry} which is provided.
        * <p>
        * Stores that have this characteristic must override the {@link #purgeExpired()} method.
        */
       EXPIRATION
    }
 
    /**
+    * The first method that will be invoked on this store to allow for the store to configure and do any additional
+    * steps, such as connecting via socket or opening file descriptors.
```

Comment:
"The first method that will be invoked to allow the store to be configured and for any additional steps, such as connecting via a socket or opening file descriptors, to be performed.